x-pack/legacy/plugins/infra/common/http_api/logs/summary.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as rt from 'io-ts';
+
+export const LOGS_SUMMARY_PATH = '/api/logs/summary';
+export const LOGS_SUMMARY_HIGHLIGHTS_PATH = '/api/logs/summary/highlights';
```

Comment:
What do you think about keeping the request handlers on the same level of the URL hierarchy, e.g.

```suggestion
export const LOGS_SUMMARY_HIGHLIGHTS_PATH = '/api/logs/summary_highlights';
```

Having leaf nodes on different levels of the hierarchy feels unnecessarily nested to me. It would also prevent us from adding url parameters to the `summary` route later on.

Reply:
My reasoning was that summary highlights are just a specialised version of highlights, therefore the nesting. But... I was wondering when I built this if we need two endpoints at all.

The request is practically the same, with the only addition of the `highlightTerms` parameter. There are some differences in the responses but they can be ironed out.

The client will still make two requests, both to the same endpoint. One without the `highlightTerms`, and one with it.

Makes sense? Do you see any downsides?

Reply:
The only downside would be that the request, response and route handler implementation become more complex. But that's hard to evaluate without a specific comparison. Maybe that's something for a follow-up though.

My comment aimed more at keeping the URL hierarchy flat and flexible.

Reply:
I'll change the endpoint. It doesn't matter one way or another :)